### PR TITLE
Add parser to find existing class details for non-destructive baking

### PR DESCRIPTION
**Pull request by othercorey**
This doesn't populate the attributes array until I decide what we need from them.

The code and docblocks remain indented. The expected indent size is 4 spaces. We can add more validation to the file formatting as needed.

### CODE REVIEW

**Comment by markstory** [comment]
Pretty neat stuff. Is the plan to use have bake generate the file as it does today with twig templates into a temporary location and then parse both the newly generated file and the current file and 'merge' the two structures together? With the merged structure I'm assuming we'll then create a dumper for the parsed class/method structure?

File: tests/TestCase/Parse/CodeParserTest.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        $this->assertSame($code, $class->methods['initialize']->code);
+
+        $doc = <<<'DOCMARKER'
+            /**
+             * Initialize method
+             *
+             * @param array $config The configuration for the Table.
+             * @return void
+             */
+        DOCMARKER;
```
Comment (markstory) [inline tests/TestCase/Parse/CodeParserTest.php:71]: Do we also preserve the file header, and any other globally defined symbols in the file? Things like constants, `require`, or other classes/functions are things I could see userland code doing.

Comment (othercorey) [reply tests/TestCase/Parse/CodeParserTest.php:71]: I'm going to add a template for the file header to allow users to customize that. We'll re-generate the file structure but allow more customization by the user.

Other code that need to be saved off are user-specific properties, but I haven't worked that out yet.

**Comment by othercorey** [comment]
> Pretty neat stuff. Is the plan to use have bake generate the file as it does today with twig templates into a temporary location and then parse both the newly generated file and the current file and 'merge' the two structures together? With the merged structure I'm assuming we'll then create a dumper for the parsed class/method structure?

I'm planning to split the twig templates into sections including file header, generated properties/methods and app-specific properties/methods.

I will merge the generated use statements and what I can detect of app-specific use statements into a sorted list.

**Pull request merged**